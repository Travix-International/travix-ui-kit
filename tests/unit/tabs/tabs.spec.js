import { shallow, mount } from 'enzyme';
import React from 'react';
import Tabs from '../../../components/tabs/tabs';
import Tab from '../../../components/tabs/tab';

describe('Tabs', () => {
  describe('#render()', () => {
    const onChange = () => {};

    it('should render correct component with basic data', () => {
      const wrapper = shallow(
        <Tabs name="tabs1" onChange={onChange}>
          <Tab title="One" />
          <Tab title="Two" />
        </Tabs>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should change "activeTab" after receive new props when "activeTab" was provided', () => {
      const wrapper = shallow(
        <Tabs name="tabs1" onChange={onChange}>
          <Tab title="One" />
          <Tab title="Two" />
        </Tabs>
      );

      expect(wrapper).toMatchSnapshot();
      wrapper.setProps({ activeTab: '1' });
      expect(wrapper).toMatchSnapshot();

      wrapper.setProps({ activeTab: '1' });
      expect(wrapper).toMatchSnapshot();
    });

    it('should not render correct with other tags', () => {
      const wrapper = shallow(
        <Tabs name="tabs1" onChange={onChange}>
          <div>test</div>
        </Tabs>
      );

      expect(wrapper.type()).toEqual(null);
    });

    it('should render correct data with content', () => {
      const wrapper = shallow(
        <Tabs name="tabs1" onChange={onChange}>
          <Tab title="One">text</Tab>
        </Tabs>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render correct data with custome values', () => {
      const wrapper = shallow(
        <Tabs name="tabs1" onChange={onChange}>
          <Tab title="One" value="1">text</Tab>
        </Tabs>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render correct data with init value', () => {
      const wrapper = shallow(
        <Tabs activeTab="1" name="tabs1" onChange={onChange}>
          <Tab title="One" value="1">text</Tab>
        </Tabs>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should call onChange method', () => {
      const renderTree = mount(
        <Tabs activeTab="1" name="tabs1" onChange={onChange}>
          <Tab title="One" value="1">text</Tab>
        </Tabs>
      );

      const tab = renderTree.find('.ui-tab');
      tab.simulate('click');

      expect(renderTree).toMatchSnapshot();
    });

    it('should call onClick method', () => {
      const renderTree = mount(
        <Tabs activeTab="1" name="tabs1">
          <Tab onClick={onChange} title="One" value="1">text</Tab>
        </Tabs>
      );

      const tab = renderTree.find('.ui-tab');
      tab.simulate('click');

      expect(renderTree).toMatchSnapshot();
    });

    it('should not call onClick method', () => {
      const renderTree = mount(<Tab title="One" value="1">text</Tab>);

      const tab = renderTree.find('.ui-tab');
      tab.simulate('click');

      expect(renderTree).toMatchSnapshot();
    });

    it('should render with custom tab content', () => {
      const wrapper = shallow(<Tab title={(<div>abc</div>)} value="1">text</Tab>);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
