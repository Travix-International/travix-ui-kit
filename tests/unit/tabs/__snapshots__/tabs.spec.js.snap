// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs #render() should call onChange method 1`] = `
<Tabs
  activeTab="1"
  name="tabs1"
  onChange={[Function]}
  open={false}
  selectionType="normal"
>
  <div
    activeTab="1"
    className="ui-tabs"
    name="tabs1"
    onChange={[Function]}
    open={false}
    selectionType="normal"
  >
    <div
      className="ui-tabs__navigation"
      role="tablist"
    >
      <Tab
        active={true}
        key="0"
        name="tabs1-1"
        onClick={[Function]}
        title="One"
        value="1"
      >
        <div
          aria-controls="tabs1-1-panel"
          aria-selected={true}
          className="ui-tab ui-tab_active"
          id="tabs1-1"
          onClick={[Function]}
          role="tab"
          tabIndex="0"
          title="One"
        >
          <div
            className="ui-tab__shadow-wrapper"
          >
            <div
              className="ui-tab__content ui-tab__content_active ui-tab__content-paddings"
            >
              One
            </div>
          </div>
        </div>
      </Tab>
    </div>
    <div
      className="ui-tabs__content"
    >
      <div
        aria-hidden={false}
        aria-labelledby="tabs1-1"
        className="ui-tabs__content-item ui-tabs__content-item_active"
        id="tabs1-1-panel"
        key="0"
        role="tabpanel"
      >
        text
      </div>
    </div>
  </div>
</Tabs>
`;

exports[`Tabs #render() should call onClick method 1`] = `
<Tabs
  activeTab="1"
  name="tabs1"
  open={false}
  selectionType="normal"
>
  <div
    activeTab="1"
    className="ui-tabs"
    name="tabs1"
    open={false}
    selectionType="normal"
  >
    <div
      className="ui-tabs__navigation"
      role="tablist"
    >
      <Tab
        active={true}
        key="0"
        name="tabs1-1"
        onClick={[Function]}
        title="One"
        value="1"
      >
        <div
          aria-controls="tabs1-1-panel"
          aria-selected={true}
          className="ui-tab ui-tab_active"
          id="tabs1-1"
          onClick={[Function]}
          role="tab"
          tabIndex="0"
          title="One"
        >
          <div
            className="ui-tab__shadow-wrapper"
          >
            <div
              className="ui-tab__content ui-tab__content_active ui-tab__content-paddings"
            >
              One
            </div>
          </div>
        </div>
      </Tab>
    </div>
    <div
      className="ui-tabs__content"
    >
      <div
        aria-hidden={false}
        aria-labelledby="tabs1-1"
        className="ui-tabs__content-item ui-tabs__content-item_active"
        id="tabs1-1-panel"
        key="0"
        role="tabpanel"
      >
        text
      </div>
    </div>
  </div>
</Tabs>
`;

exports[`Tabs #render() should change "activeTab" after receive new props when "activeTab" was provided 1`] = `
<div
  activeTab="0"
  className="ui-tabs"
  name="tabs1"
  onChange={[Function]}
  open={false}
  selectionType="normal"
>
  <div
    className="ui-tabs__navigation"
    role="tablist"
  >
    <Tab
      active={true}
      key="0"
      name="tabs1-0"
      onClick={[Function]}
      title="One"
      value="0"
    />
    <Tab
      active={false}
      key="1"
      name="tabs1-1"
      onClick={[Function]}
      title="Two"
      value="1"
    />
  </div>
</div>
`;

exports[`Tabs #render() should change "activeTab" after receive new props when "activeTab" was provided 2`] = `
<div
  activeTab="1"
  className="ui-tabs"
  name="tabs1"
  onChange={[Function]}
  open={false}
  selectionType="normal"
>
  <div
    className="ui-tabs__navigation"
    role="tablist"
  >
    <Tab
      active={false}
      key="0"
      name="tabs1-0"
      onClick={[Function]}
      title="One"
      value="0"
    />
    <Tab
      active={true}
      key="1"
      name="tabs1-1"
      onClick={[Function]}
      title="Two"
      value="1"
    />
  </div>
</div>
`;

exports[`Tabs #render() should change "activeTab" after receive new props when "activeTab" was provided 3`] = `
<div
  activeTab="1"
  className="ui-tabs"
  name="tabs1"
  onChange={[Function]}
  open={false}
  selectionType="normal"
>
  <div
    className="ui-tabs__navigation"
    role="tablist"
  >
    <Tab
      active={false}
      key="0"
      name="tabs1-0"
      onClick={[Function]}
      title="One"
      value="0"
    />
    <Tab
      active={true}
      key="1"
      name="tabs1-1"
      onClick={[Function]}
      title="Two"
      value="1"
    />
  </div>
</div>
`;

exports[`Tabs #render() should not call onClick method 1`] = `
<Tab
  active={false}
  name=""
  title="One"
  value="1"
>
  <div
    aria-controls="-panel"
    aria-selected={false}
    className="ui-tab"
    id=""
    onClick={[Function]}
    role="tab"
    tabIndex="0"
    title="One"
  >
    <div
      className="ui-tab__shadow-wrapper"
    >
      <div
        className="ui-tab__content ui-tab__content-paddings"
      >
        One
      </div>
    </div>
  </div>
</Tab>
`;

exports[`Tabs #render() should render correct component with basic data 1`] = `
<div
  activeTab="0"
  className="ui-tabs"
  name="tabs1"
  onChange={[Function]}
  open={false}
  selectionType="normal"
>
  <div
    className="ui-tabs__navigation"
    role="tablist"
  >
    <Tab
      active={true}
      key="0"
      name="tabs1-0"
      onClick={[Function]}
      title="One"
      value="0"
    />
    <Tab
      active={false}
      key="1"
      name="tabs1-1"
      onClick={[Function]}
      title="Two"
      value="1"
    />
  </div>
</div>
`;

exports[`Tabs #render() should render correct data with content 1`] = `
<div
  activeTab="0"
  className="ui-tabs"
  name="tabs1"
  onChange={[Function]}
  open={false}
  selectionType="normal"
>
  <div
    className="ui-tabs__navigation"
    role="tablist"
  >
    <Tab
      active={true}
      key="0"
      name="tabs1-0"
      onClick={[Function]}
      title="One"
      value="0"
    />
  </div>
  <div
    className="ui-tabs__content"
  >
    <div
      aria-hidden={false}
      aria-labelledby="tabs1-0"
      className="ui-tabs__content-item ui-tabs__content-item_active"
      id="tabs1-0-panel"
      key="0"
      role="tabpanel"
    >
      text
    </div>
  </div>
</div>
`;

exports[`Tabs #render() should render correct data with custome values 1`] = `
<div
  activeTab="0"
  className="ui-tabs"
  name="tabs1"
  onChange={[Function]}
  open={false}
  selectionType="normal"
>
  <div
    className="ui-tabs__navigation"
    role="tablist"
  >
    <Tab
      active={false}
      key="0"
      name="tabs1-1"
      onClick={[Function]}
      title="One"
      value="1"
    />
  </div>
  <div
    className="ui-tabs__content"
  >
    <div
      aria-hidden={true}
      aria-labelledby="tabs1-1"
      className="ui-tabs__content-item"
      id="tabs1-1-panel"
      key="0"
      role="tabpanel"
    >
      text
    </div>
  </div>
</div>
`;

exports[`Tabs #render() should render correct data with init value 1`] = `
<div
  activeTab="1"
  className="ui-tabs"
  name="tabs1"
  onChange={[Function]}
  open={false}
  selectionType="normal"
>
  <div
    className="ui-tabs__navigation"
    role="tablist"
  >
    <Tab
      active={true}
      key="0"
      name="tabs1-1"
      onClick={[Function]}
      title="One"
      value="1"
    />
  </div>
  <div
    className="ui-tabs__content"
  >
    <div
      aria-hidden={false}
      aria-labelledby="tabs1-1"
      className="ui-tabs__content-item ui-tabs__content-item_active"
      id="tabs1-1-panel"
      key="0"
      role="tabpanel"
    >
      text
    </div>
  </div>
</div>
`;

exports[`Tabs #render() should render with custom tab content 1`] = `
<div
  aria-controls="-panel"
  aria-selected={false}
  className="ui-tab"
  id=""
  onClick={[Function]}
  role="tab"
  tabIndex="0"
  title=""
>
  <div
    className="ui-tab__shadow-wrapper"
  >
    <div
      className="ui-tab__content"
    >
      <div>
        abc
      </div>
    </div>
  </div>
</div>
`;
